analysis/src/python/data_analysis/analysis/timing/time_compare.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import logging
+import os
+import sys
+import time
+from pathlib import Path
+from typing import List
+
+import numpy as np
+import pandas as pd
+from hyperstyle.src.python.review.application_config import LanguageVersion
+from hyperstyle.src.python.review.reviewers.common import LANGUAGE_TO_INSPECTORS
+
+from analysis.src.python.data_analysis.model.column_name import SubmissionColumns
+from analysis.src.python.data_analysis.utils.df_utils import read_df
+from analysis.src.python.data_analysis.utils.logging_utils import configure_logger
+from analysis.src.python.evaluation.common.file_util import create_file
+from analysis.src.python.evaluation.common.parallel_util import run_and_wait
+
+TEMPLATE_FOLDER = Path(__file__).parents[4] / 'resources' / 'evaluation' / 'qodana' / 'project_templates'
+PROFILE_FOLDER = Path(__file__).parents[4] / 'resources' / 'evaluation' / 'qodana' / 'inspection_profiles'
+JAVA_QODANA_IMAGE_PATH = 'jetbrains/qodana'
+PYTHON_QODANA_IMAGE_PATH = 'jetbrains/qodana-python'
+
+
+def write_code_to_template(language: LanguageVersion, code: str):
+    if language.is_java():
+        project_dir = TEMPLATE_FOLDER / 'java'
+        main_path = project_dir / 'src/main/java/Main.java'
+    elif language == LanguageVersion.PYTHON_3:
+        project_dir = TEMPLATE_FOLDER / 'python'
+        main_path = project_dir / 'main.py'
+    else:
+        raise NotImplementedError(f'{language} needs implementation.')
+
+    create_file(main_path, code)
+    return project_dir, main_path
+
+
+def get_qodana_configuration(language: LanguageVersion):
+    if language.is_java():
+        qodana_image_path = 'jetbrains/qodana'
+        profile_path = PROFILE_FOLDER / 'java_profile.xml'
+    elif language == LanguageVersion.PYTHON_3:
+        qodana_image_path = 'jetbrains/qodana-python'
+        profile_path = PROFILE_FOLDER / 'python_profile.xml'
+    else:
+        raise NotImplementedError(f'{language} needs implementation.')
+
+    return qodana_image_path, profile_path
+
+
+def run_hyperstyle(language: str, code: str, repeat: int = 5):
+    language_version = LanguageVersion.from_value(language)
+
+    language = LanguageVersion.from_value(language)
+    inspectors = LANGUAGE_TO_INSPECTORS.get(language, [])
+
+    inspectors_config = {
+        'language_version': language_version,
+        'n_cpu': 1,
+    }
+
+    project_dir, main_path = write_code_to_template(language_version, code)
+    repeat_times = []
+
+    for i in range(repeat):
+        logging.info(f'Hyperstyle qodana iteration {i}-th')
+
+        start_time = time.time()
+        for inspector in inspectors:
+            inspector.inspect(main_path, inspectors_config)
+        end_time = time.time()
+        repeat_times.append(end_time - start_time)
```

Comment:
Can we add a function to measure time and reuse in two places to send a different part?